open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidationUtils.java

```diff
@@ -0,0 +1,132 @@
+     *
+     * @param consentManageData Consent Manage Data
+     * @return Whether the required parameters are present
+     */
+    static boolean isMandatoryParamsPresent(ConsentManageData consentManageData, String idempotencyKeyName) {
+        String idempotencyKeyValue = consentManageData.getHeaders().get(idempotencyKeyName);
+        return idempotencyKeyValue != null && consentManageData.getPayload() != null &&
+                consentManageData.getClientId() != null;
+    }
+
+    /**
+     * Method to check whether Idempotency handling is required.
+     *
+     * @return True if idempotency is required, else False.
+     */
+    static boolean isIdempotencyEnabledFromConfig() {
+        return parser.isIdempotencyValidationEnabled();
+    }
+
+    /**
+     * Method to retrieve the consent ids that have the idempotency key name and value as attribute.
+     *
+     * @param idempotencyKeyName     Idempotency Key Name
+     * @param idempotencyKeyValue    Idempotency Key Value
+     * @return   List of consent ids
+     */
+    static ArrayList<String> getConsentIdsFromIdempotencyKey(String idempotencyKeyName,
+                                                             String idempotencyKeyValue) {
+        try {
+            return consentCoreService.getConsentIdByConsentAttributeNameAndValue(
+                    idempotencyKeyName, idempotencyKeyValue);
+        } catch (ConsentManagementException e) {
+            log.debug("No consent ids found for the idempotency key value");
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * Method to compare the client ID sent in the request and client id retrieved from the database.
+     *
+     * @param requestClientID     Client ID sent in the request
+     * @param dbClientId          client ID retrieved from the database
+     * @return   Whether JSON client Ids are equal
+     */
+    static boolean isClientIdsMatching(String requestClientID, String dbClientId) {
+        return requestClientID.equals(dbClientId);
+    }
+
+    /**
+     * Method to check whether difference between two dates is less than the configured time.
+     *
+     * @param createdTime Created Time of the request
+     * @return  Whether the request is received within allowed time
+     */
+    static boolean isRequestReceivedWithinAllowedTime(long createdTime) {
+
+        if (createdTime == 0L) {
+            return false;
+        }
+        String allowedTimeDuration = parser.getIdempotencyAllowedTime();
+        if (allowedTimeDuration != null) {
+            OffsetDateTime createdDate = OffsetDateTime.parse(convertToISO8601DateTimeFormat(createdTime));
+            OffsetDateTime currDate = OffsetDateTime.now(createdDate.getOffset());
+
+            long diffInHours = Duration.between(createdDate, currDate).toMinutes();
+            return diffInHours <= Long.parseLong(allowedTimeDuration);
+        } else {
+            log.error("Idempotency Allowed duration is null");
```

Comment:
Shall we say that the error is due to not configuring the 'Idempotency Allowed Duration' in the configs correctly?

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/51/commits/e1205691228fc4b76853a0eab7dd99dac2516661